tachyon/math/polynomials/combination/linear_combination_term.h

```diff
@@ -0,0 +1,89 @@
+#ifndef TACHYON_MATH_POLYNOMIALS_COMBINATION_LINEAR_COMBINATION_TERM_H_
+#define TACHYON_MATH_POLYNOMIALS_COMBINATION_LINEAR_COMBINATION_TERM_H_
+
+#include <functional>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/buffer/copyable.h"
+#include "tachyon/base/parallelize.h"
+
+namespace tachyon {
+namespace math {
+
+template <typename F>
+struct LinearCombinationTerm {
+  F coefficient;
+  std::vector<size_t> indexes;
+
+  template <typename MLE, typename Container>
+  F Evaluate(
+      const Container& point,
+      const std::vector<std::shared_ptr<MLE>>& flattened_ml_evaluations) const {
+    // NOTE(ashjeong): referenced multivariate_sparse_coefficients.h
+#if defined(TACHYON_HAS_OPENMP)
+    std::vector<F> results = base::ParallelizeMap(
+        indexes,
+        [&flattened_ml_evaluations, &point](absl::Span<const size_t> chunk) {
+          return EvaluateSerial(chunk, flattened_ml_evaluations, point);
+        });
+    return coefficient * std::accumulate(results.begin(), results.end(),
+                                         F::One(), std::multiplies<>());
+#else
+    return coefficient *
+           EvaluateSerial(indexes, flattened_ml_evaluations, point);
+#endif
+  }
+
+  bool operator==(const LinearCombinationTerm& other) const {
+    return coefficient == other.coefficient && indexes == other.indexes;
+  }
+  bool operator!=(const LinearCombinationTerm& other) const {
+    return !operator==(other);
+  }
+
+ private:
+  template <typename MLE, typename Container>
+  static F EvaluateSerial(
+      absl::Span<const size_t> indexes,
+      const std::vector<std::shared_ptr<MLE>>& flattened_ml_evaluations,
+      const Container& point) {
```

Comment:
In my opinion, it's good to move `indexes` parameter to the end for readability considering the function above.